### ISSUE TITLE
[Issue] Adding environment variables while `azd up`-ing?

### ISSUE THREAD

**Issue by justinyoo** [Azure/azure-dev#2989]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install) 👉 v1.4.5

**Output from `azd version`**

```bash
azd version 1.4.5 (commit 6cb723e44eb0697d4d8d09f7694169f5410c7d6f)
```

**Describe the bug**

It's applicable for .NET Aspire.

When I run `azd infra synth`, it generates the manifest file that defines some environment variables by default. That means, if I run `azd up` those pre-defined environment variables are deployed.

If I want to add new environment variables defined in `appsettings.json` or `secrets.json` what will be the best bet? What I'm currently doing is that:

1. Run `azd infra synth`
1. Update the manifest file by adding new environment variables
1. Run `azd up`

But there might be a simpler way.

**To Reproduce**

N/A

**Expected behavior**

N/A

**Environment**
Information on your environment:
    * Language name and version: C# (.NET 8)
    * IDE and version : Visual Studio 2022, v17.9

**Additional context**

N/A

**Comment by jongio** [comment]
Have you looked into implementing a custom hook?  https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/azd-extensibility

**Comment by rajeshkamal5050** [comment]
@ellismg to check with @justinyoo on the use-case. Also, check with Aspire and/or .NET folks for inputs.

**Comment by justinyoo** [comment]
These are potentially related to this issue:

- AZURE_LOCATION error: #3031 
- Expose to internet: #3027

**Comment by justinyoo** [comment]
@timheuer Yep, I found a clue how to make use of environment variables and pass them from AppHost to individual applications from the [aspire-samples](https://github.com/dotnet/aspire-samples) repo.

**Comment by rajeshkamal5050** [comment]
@justinyoo can we close this issue?
